src/BookReader.js

```diff
@@ -847,8 +847,12 @@ BookReader.prototype.zoom = function(direction) {
     }
     break;
   case this.constModeThumb:
-    // XXX update zoomThumb for named directions
-    this.zoomThumb(direction);
+    if (direction == 1) {
+    // XXX other cases
+      this.zoomThumb('in');
+    } else {
+      this.zoomThumb('out');
+    }
     break;
```

Comment:
Woohoo! Now these are all the same, we can replace the switch with `this.activeMode.zoom(direction)` ; I'd also change this function to only take in `in` / `out`.

Note these are breaking changes to public APIs; that's fine, just mention in the commit message/changelog/releaselog :+1:

Reply:
`activeMode`?
I'll have to go back and understand this

Reply:
It's defined in this file; `activeMode` is the... active mode! :P Type: `Mode1Up | Mode2Up | ModeThumb` . Since all of those classes implement `zoom(dir: 'in' | 'out')` , we can call `.zoom` on `activeMode` !